lib/src/folding/parsers/spaces_foldable_block_parser.dart

```diff
@@ -0,0 +1,109 @@
+import 'dart:collection';
+
+import 'package:charcode/ascii.dart';
+import '../foldable_block.dart';
+import '../foldable_block_type.dart';
+
+import 'abstract.dart';
+
+///A parser for foldable blocks from spaces count at start of each line.
+class SpacesFoldableBlockParser extends AbstractFoldableBlockParser {
+  static const _kSeparatorLine = -1;
+
+  void parse(String text) {
+    _addWhitespacesBlocks(text);
+    finalize();
+  }
+
+  void _addWhitespacesBlocks(String text) {
```

Comment:
This is too long a method and should be split.
You can accumulate open blocks in a field because this class is stateful anyway.
Good extraction candidates are:
`void _openBlock(int line, int indent)`
`void _closeBlocks(int line, int minIndent)`